maas/client/viscera/machines.py

```diff
@@ -118,7 +118,9 @@ class MachinesType(NodesType):
             interfaces: typing.Sequence[InterfaceParam]=None,
             memory: float=None,
             pod: str=None,
+            not_pod: str=None,
```

Comment:
For both `pod` and `not_pod` you should accept your new `Pod` class you created in the other branch. That provides a better experience then always requiring the call to convert that to a string.